webapp/channels/src/components/tabs/tabs/index.tsx

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import type {TabsProps} from 'react-bootstrap';
+import {Tabs as ReactBootstrapTabs} from 'react-bootstrap';
+
+import './style.scss';
+
+type Props = TabsProps & {
+    children?: React.ReactNode;
+}
```

Comment:
I wouldn't expose all the props here. Some of the props we are always passing the same value (`mountOnEnter`, `unmountOnExit`), and there are plenty of props we are not really using.

Having a smaller interface helps when we decide to substitute these tabs to something else, since there are less things to support on the new implementation.

Also, you give less options (and therefore, less decisions) to future developers.

Reply:
We, (or rather I) have faced this problem countless times where we had a wrapper over a library component ad we decided to expose only a subset of props, making it difficult to do several things which the underlying component allowed but we decided our wrapper should now. This has caused more problems than solved IMHO and we should not do this. When, and if, we decide to get rid of react-bootstrap, which I don't see happening soon, we can deal with it.

I'm 4/5 we should expose all underlying props.

Reply:
My position on this is that the least amount of interface, the least cognitive load and the least complexity we have. That way we can test the things we care about, and the rest will always "stay the same". If we were going to add test to this component, we won't test all the props expose from the library, and therefore, whenever someone comes up with using a different prop we usually don't use, things can get hairy.

And whenever we need something extra, we can always expose it. Just because the wrapper doesn't allow it now, does not mean it cannot allow it ever. We just need to consciously decide what to open, when, and how.

Nevertheless, as of today, I still don't have a strong opinion on this, so I will leave @hmhealey chime in in case he has a stronger opinion (on either direction). But happy to leave it as it is today.

Reply:
Done.